tests/ttnn/unit_tests/operations/test_moreh_clip_grad_norm.py

```diff
@@ -0,0 +1,224 @@
+            input_shapes.append(input_shape)
+
+        npu_total_norm = ttnn.experimental.operations.primary.moreh_clip_grad_norm_(npu_inputs, max_norm, norm_type)
+        new_npu_total_norm = ttnn.operations.moreh.clip_grad_norm(new_npu_inputs, max_norm, norm_type)
+        # reshape to flatten due to mismatch shape
+        expected_total_norm = to_cpu(npu_total_norm, [1, 1, 1, 1]).reshape(-1)
+        actual_total_norm = to_cpu(new_npu_total_norm, [1, 1, 1, 1]).reshape(-1)
+        assert torch.equal(expected_total_norm, actual_total_norm)
+
+        # Check inputs
+        for i in range(num_parameters):
+            expected_input_i = ttnn.to_torch(npu_inputs[i])
+            actual_input_i = ttnn.to_torch(new_npu_inputs[i])
+            assert torch.equal(expected_input_i, actual_input_i)
+
+
+@pytest.mark.skip(reason="assertion fails during binary op input shape comparison because of different padding")
+@pytest.mark.parametrize("num_iters_of_each_case", [2])
+@pytest.mark.parametrize("range_of_padding", [(0, 21, 10)])  # [0, 10, 20]
+@pytest.mark.parametrize("range_of_n", [(1, 4)])
+@pytest.mark.parametrize("range_of_c", [(1, 4)])
+@pytest.mark.parametrize("range_of_ht", [(1, 4)])
+@pytest.mark.parametrize("range_of_wt", [(1, 4)])
+@pytest.mark.parametrize("max_norm", [2.0, 1.0, -1.0])
+@pytest.mark.parametrize("norm_type", [2.0, -0.8, 2.2])
+@pytest.mark.parametrize("num_parameters", [32, 128])
+def test_moreh_clip_grad_norm(
+    num_iters_of_each_case,
+    num_parameters,
+    max_norm,
+    norm_type,
+    range_of_n,
+    range_of_c,
+    range_of_ht,
+    range_of_wt,
+    range_of_padding,
+    device,
+):
+    torch.manual_seed(2023)
+    random.seed(2023)
+
+    cpu_dtype = torch.float32
+    npu_dtype = ttnn.bfloat16
+
+    cpu_inputs = []
+    npu_inputs = []
+    input_shapes = []
+
+    for _ in range(num_iters_of_each_case):
+        for _ in range(num_parameters):
+            n = random.randint(*range_of_n)
+            c = random.randint(*range_of_c)
+            ht = random.randint(*range_of_ht)
+            wt = random.randint(*range_of_wt)
+            padding_h = random.randrange(*range_of_padding)
+            padding_w = random.randrange(*range_of_padding)
+
+            input_shape = (
+                n,
+                c,
+                ht * TILE_HEIGHT - padding_h,
+                wt * TILE_WIDTH - padding_w,
+            )
+
+            param = torch.nn.Parameter(torch.empty(input_shape, dtype=cpu_dtype))
+            grad = torch.empty(input_shape, dtype=cpu_dtype).uniform_(0, 2.5)
+            param.grad = grad
+
+            cpu_inputs.append(param)
+            npu_inputs.append(to_npu(grad.clone().bfloat16(), device, npu_dtype=npu_dtype))
+            input_shapes.append(input_shape)
+
+        cpu_total_norm = torch.nn.utils.clip_grad_norm_(cpu_inputs, max_norm, norm_type)
+        npu_total_norm = ttnn.operations.moreh.clip_grad_norm(npu_inputs, max_norm, norm_type)
```

Comment:
```suggestion
        npu_total_norm = ttnn.moreh.clip_grad_norm(npu_inputs, max_norm, norm_type)
```
I think `operations.` is not neessary

Reply:
I will refactor the pytest when we fix the issue with the precision in later commit.